exercise/submission_models.py

```diff
@@ -30,7 +31,8 @@ def get_queryset(self):
     def create_from_post(self, exercise, submitters, request):
         new_submission = Submission.objects.create(
             exercise=exercise,
-            submission_data=query_dict_to_list_of_tuples(request.POST)
+            submission_data=query_dict_to_list_of_tuples(request.POST),
+            meta_data=json.loads(request.POST.get('__aplus__','{}'))
```

Comment:
Please add the comma so the next change doesn't need to do the same as this one did, i.e, edit the line above just to add the comma.

Also, there should be space between arguments, please check [spaces section of PEP 8](https://www.python.org/dev/peps/pep-0008/#whitespace-in-expressions-and-statements).

Lastly, this should handle invalid JSON some way too, mostly so that the user doesn't get 500 for invalid json in `__aplus__`.

Reply:
Should meta_data just be set to an empty dict continue grading or should there be some message returned  if there is invalid json

Reply:
About potential JSON syntax errors in the meta data: ignoring errors silently makes debugging harder, so we could at least log an error. Since the meta data is not critical for grading, it could be skipped and the submission could still be sent to the grader, but it is also suspicious. Is there another problem in the submission if the client sends broken JSON with it? Well, what does Jaakko think?

Reply:
@raphendyr

Reply:
At least the API should return error on invalid POST. Web side requires some UI to handle that though. Luckily, that shouldn't happen there.